cpp/src/slice2cpp/Gen.cpp

```diff
@@ -2493,22 +2696,24 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& p)
         returns.push_back(futureDoc);
         writeOpDocSummary(H, p, comment, OpDocInParams, false, StringList(), postParams, returns);
     }
-    H << nl << "template<template<typename> class P = ::std::promise>";
-    H << nl << deprecateSymbol << "auto " << name << "Async" << spar << inParamsDecl << contextDecl << epar;
-    H.inc();
-    H << nl << "-> decltype(::std::declval<P<" << futureT << ">>().get_future())";
-    H.dec();
-    H << sb;
 
-    H << nl << "return _makePromiseOutgoing<" << futureT << ", P>" << spar;
+    H << nl << deprecateSymbol << "std::future<" << futureT << "> " << name << "Async" << spar << inParamsDecl
+        << contextDecl << epar << ";";
 
-    H << "false, this" << string("&" + interface->name() + "Prx::_iceI_" + name);
+    C << sp;
+    C << nl << "std::future<" << futureTAbsolute << ">";
+    C << nl;
+    C << scoped << name << "Async" << spar << inParamsImplDecl << "const ::Ice::Context& context" << epar;
+
+    C << sb;
+    C << nl << "return _makePromiseOutgoing<" << futureT << ", std::promise>" << spar;
```

Comment:
Here too, should it be `::std::promise`?

Reply:
Fixed. Naturally this is temporary as we should fix _makePromiseOutgoing to always use std::promise.